R/deps.R

```diff
@@ -236,14 +239,17 @@ UNAVAILABLE <- 2L
 
 update.package_deps <- function(object,
                            dependencies = NA,
-                           upgrade = TRUE,
+                           upgrade = NA,
```

Comment:
How about having  `"ask"` instead of  `NA`? And the default could be `if  (interactive()) "ask" else TRUE`. That would allow us to have other "modes" later, e.g. `"cran-only"`, etc. and it is also more readable. In `install.packages()` I have to check the docs every single time when I see `dependencies = NA`  or `NULL` or  `TRUE`, etc.

Reply:
I used NA so it would take only logical inputs, but I suppose the distinction is minor.

Reply:
Yeah, I think in languages with constants, this would be a constant or enum, which in R is basically a string. `NA` is  good to make it logical, but I think it is just cleaner to make it a "constant", that is also extendable.

Reply:
I mostly agree with you I think in this case, I posted on slack to see if anyone else has opinions.